engine/src/main/java/org/terasology/world/propagation/BatchPropagator.java

```diff
@@ -55,21 +45,86 @@
      * @param pos   The position of the block
      * @param block The block type
      */
-    void propagateFrom(Vector3i pos, Block block);
+    void propagateFrom(Vector3ic pos, Block block);
+
+    @Deprecated
+    default void propagateFrom(Vector3i pos, Block block) {
+        propagateFrom(JomlUtil.from(pos), block);
+    };
 
     /**
      * Propagates a specific value out from a location
      *
      * @param pos   The position to propagate out of
      * @param value The value to propagate out
      */
-    void propagateFrom(Vector3i pos, byte value);
+    void propagateFrom(Vector3ic pos, byte value);
+
+    @Deprecated
+    default void propagateFrom(Vector3i pos, byte value) {
+        propagateFrom(JomlUtil.from(pos), value);
+    }
 
     /**
      * TODO: Document
      *
      * @param pos   The position to regenerate at
      * @param value The original value at this position
      */
-    void regenerate(Vector3i pos, byte value);
+    void regenerate(Vector3ic pos, byte value);
+
+    @Deprecated
+    default void regenerate(Vector3i pos, byte value) {
+        regenerate(JomlUtil.from(pos), value);
+    }
+
+    /**
+     * Populates a target array with the minimum value adjacent to each location, including the location itself.
+     *
+     * @param source
+     * @param target
+     * @param populateMargins Whether to populate the edges of the target array
+     */
+    default void populateMinAdjacent2D(int[] source, int[] target, int dimX, int dimY, boolean populateMargins) {
```

Comment:
I was thinking static but this might work better. allows for different implementations. :+1:

Reply:
I don't know whether we actually want different implementations? Maybe we should keep it static and only open that up if we actually find a use case for that?